app/controllers/buyers/bulk_base_controller.rb

```diff
@@ -4,37 +4,31 @@ class Buyers::BulkBaseController < FrontendController
   before_action :authorize_bulk_operations
   before_action :initializa_errors, only: :create
 
+  after_action :handle_errors, only: :create
+
   def new; end
 
   def create
-    @errors = nil
-
     recipients.each do |recipient|
       message = current_account.messages.build send_emails_params
       message.to = recipient
 
       @errors << message unless message.save && message.deliver!
     end
-
-    handle_errors
   end
 
   protected
 
   def initializa_errors
-    @errors = nil
+    @errors = []
   end
 
   def authorize_bulk_operations
     authorize! :manage, scope
   end
 
-  def permitted_params
-    params.permit(selected: [], send_emails: %i[subject body])
-  end
-
   def send_emails_params
-    permitted_params.fetch(:send_emails)
+    params.permit(send_emails: %i[subject body]).fetch(:send_emails)
```

Comment:
Looks great! Just one question - how did you know only `subject` and  `body` are supported?

Reply:
```suggestion
    params.require(:send_emails).permit(:subject, :body)
```

Reply:
Looking at the form in the template probably

Reply:
That's good, just asking. Also if we can have a simple test for these `@errors`, would be good as it is easy to overlook something and break them otherwise.

Reply:
The simple test for @errors turned up in an important cleanup/refactoring. But I think the outcome is much better than what we currently have.